doc/syntax/comments.rdoc

```diff
@@ -35,3 +35,188 @@ syntax error:
     Will not work
     =end
   end
+
+= Magic Comments
+
+While comments are typically ignored by Ruby, special "magic comments" contain
+directives that affect how the code is interpreted.
+
+Top-level magic comments must start on the first line, or on the second line if
+the first line looks like `#! shebang line`.
+
+NOTE: Magic comments affect only the file in which they appear;
+other files are unaffected.
+
+== Alternative syntax
+
+Magic comment may consist of a single directive multiple (as in the examples above).
+Alternatively, multiple directives may appear on the same line if separated by ";"
+and wrapped between "-*-" (See Emacs' file variables: https://www.gnu.org/software/emacs/manual/html_node/emacs/Specifying-File-Variables.html)
+
+  # -*- emacs-compatible; encoding: big5; frozen_string_literal: true -*-
+
+  p 'hello'.frozen? # => true
+  p 'hello'.encoding # => true
+
+== <code>encoding</code> Directive
+
+Indicates which string encoding should be used for string literals:
+
+  # encoding: big5
+
+  ''.encoding # => #<Encoding:Big5>
+
+Default encoding is UTF-8.
+
+It must appear in the first comment section of a file
+
+The word "coding" may be used instead of "encoding".
+
+== <code>frozen_string_literal</code> Directive
+
+When appears in the top section of a file, indicates that string literals should be allocated once at parse time and frozen.
+
+  # frozen_string_literal: true
+
+  3.times do
+    p 'hello'.object_id # => prints same number
+  end
+  p 'world'.frozen? # => true
+
+The default is false. Without the comment, or with
+<code># frozen_string_literal: false</code>,
+the example above would print 3 different numbers and "false".
+
+Starting in Ruby 3.0, string literals that are dynamic are not frozen nor reused:
+
+  # frozen_string_literal: true
+
+  p "Addition: #{2 + 2}".frozen? # => false
+
+It must appear in the first comment section of a file
+
+== <code>warn_indent</code> Directive
+
+This directive can turn detection of bad indentation for statements that follow it:
+
+  def foo
+    end # => no warning
+
+  # warn_indent: true
+  def bar
+    end # => warning: mismatched indentations at 'end' with 'def' at 6
+
+Note that these warnings are issued when running with warnings (<code>ruby -w</code>), unless a directive sets this to false.
+
+== <code>shareable_constant_value</code> Directive
+
+Note: This directive is experimental in Ruby 3.0 and may change in future releases.
+
+This special directive helps to create constants that hold only immutable objects, or {Ractor-shareable}[rdoc-ref:doc/Ractor#class-Ractor-label-Shareable+and+unshareable+objects] constants.
```

Comment:
The link is not working this way... Should _theoretically_ be
```
[rdoc-ref:Ractor@Shareable+and+unshareable+objects]
```
(as per http://docs.seattlerb.org/rdoc/RDoc/Markup.html#label-Links)
...but somehow this points to `#label-Shareable+and+unshareable+objects`, while the real anchor is `#class-Ractor-label-Shareable+and+unshareable+objects`. I'll try to look into it tomorrow, but if it will stay the single unresolved issue, it is not worth it :) At least my version does renders as a link to `Ractor` class

Reply:
I didn't check it 😅